### ISSUE TITLE
Random Screen Freezing

### ISSUE THREAD

**Issue by btracz** [RealDebugMonkey/ZeachCobbler#67]
Fisrt, thank you for your powerful mod. 

Sometimes the interface freezes, most probably when splitting or exploding on a virus.
Only refreshing the page solves the issue.

To help you debug, I've noticed the following error in the javascript console :
`Uncaught IndexSizeError: Failed to execute 'arc' on 'CanvasRenderingContext2D': The radius provided (-324.016) is negative`

Has anyone experienced this behaviour lately ? It's very recent to me.

**Comment by sofiadif** [comment]
Have the very same problem.. Hope an update comes soon..
Probably duo to the new agario update..

Once the game freezes, i can stil see myself moving/splitting on the mini-map until i die

**Comment by PepinCZ** [comment]
This would be probably some agar.io update. I think we need update some function. I would try to solve it...

**Comment by PepinCZ** [comment]
Uh after playing half hour i finally stuck with no movement. I dont even see any error in console... and its random.

**Comment by ghost** [comment]
Finaly my screen gets totally stuck... Its some minimap error

**Comment by ghost** [comment]
Here you got material to work with http://pastebin.com/Q6kHtqzq
At the end of that file are errors catched in console with lines.

**Comment by vmagick59** [comment]
hello
Have the same issue, freeze and still moving on the minimap !
i ve noticed too freezing (without moving on the minimap this time) when name of players are chinese characters.
Hope it can help to fix it

Thanks for this mod its the best !

**Comment by ghost** [comment]
@vmagick59 ye it can possible be bad manipulation with some unicode characters... bud I now cant access my main PC so I cant really help.. but here http://pastebin.com/Q6kHtqzq u have material to work with, if you want to try to fix this. At end of that file are error from console with lines.

**Comment by sofiadif** [comment]
Where are the crew ? :)
Agario sucks without this mod, hope it won't be the end of this..

**Comment by mrblah1** [comment]
Having the same issue, doesn't seem to be related to splitting but just at random times the overall screen just freezes and the teams leaderboard disappears but you can still see the bot moving around on the minimap.  I also noticed that the Apos bot experiences something similar so it appears to be a change in the agar servers

**Comment by TheCranky** [comment]
Same problem here started about 2-3 days ago only way I managed to temporarily fix it while in game is to hold the f11 key. It works but its painful to watch

**Comment by mrblah1** [comment]
Ok so I know very little about programming but found a workaround that just disables the mini-map which seems to fix the issue:

Go into the tapermonkey dashboard and click on zeach cobbler which brings up the code, scroll down just a little until you see "var miniMapCtx" and on the 2nd line where it says ".appendTo(jQuery('body'))" and change the word "body" to "canvas".

I'm sure there's a better way to do it but that at least disables the mini-map and seems to fix the freezing issues.

**Comment by ghost** [comment]
@mrblah1 lol you dont disabling it, you just applying it to "canvas"...

**Comment by kauegimenes** [comment]
@PepinCZ And i believe there is no tag named `canvas`, that's why it prob works lol.

**Comment by sofiadif** [comment]
@mrblah1: Not working..

Sometimes you get lucky, being able to play for an hour.. But most times just after 10min or less

**Comment by ghost** [comment]
@sofiadif if you want to play without this shit bug just comment line in code that executing the minimap.. apply it to "canvas" isnt really good fix

**Comment by sofiadif** [comment]
@PepinCZ: I can live without the minimap, could you explain a bit more?
Maybe write the text and tell me where to paste/change it ? :)

**Comment by kaue** [comment]
Just submit a PR with a temp fix until somebody fix it the right way

**Comment by ghost** [comment]
@kauegimenes I already submitted few things but I think @RealDebugMonkey isnt there...

**Comment by ghost** [comment]
If you want to temporarily fix the issue, follow this steps: (this will actually remove minimap - that causing the error)

1) Open TamperMonkey's controll panel
2) Edit the Zeach Cobbler script
3) Go to line 1028, there will be `drawMiniMap();`
4) Just add comment before it, so it will look like this: `//drawMiniMap();`
5) Save it
6) Done

Or if you are lazy, just install this modified version of the script https://github.com/PepinCZ/ZeachCobbler/raw/minimap_fix/ZeachCobbler.user.js (it also has some changes that are in my pull request...)

**Comment by kaue** [comment]
@PepinCZ maybe @RealDebugMonkey should give permission for you to merge PRs or we could create a new fork. Almost a month to merge is sad :(

**Comment by ghost** [comment]
@kauegimenes yep, thats good idea! Also please, give me report if my temporarily fix working.

**Comment by mrblah1** [comment]
Awesome!  Thanks Pepin, I'll try it out tonight and let you know how it works.

**Comment by kaue** [comment]
@PepinCZ If you want to move forward with a new fork, i will follow you. Just change the name of the bot to prevent conflict. 
I will try your version this weekend.
Cheers

**Comment by ghost** [comment]
@kauegimenes I dont know new fork is good idea, because a lot of people will use this "outdated" repository.

**Comment by sofiadif** [comment]
@PepinCZ: Thank you, Can confirm that it works now with minimap disabled :)

**Comment by kaue** [comment]
@PepinCZ Yeah, but since it's broken. It's a possible solution.

**Comment by ghost** [comment]
Ok, I think I finaly fixed it, I just need few mins to test it.

**Comment by ghost** [comment]
So I finaly managed to fix it... It's in pull request, but instead of waiting for @RealDebugMonkey to accept PR use my fork :smiling_imp: https://github.com/PepinCZ/ZeachCobbler/raw/master/ZeachCobbler.user.js
Also please test it, and let me know if it's working! (I tested it cca. hour, but who knows :smile:)

**Comment by Aers20** [comment]
@PepinCZ I tried your raw file, but now it wont connect any servers and the moment when i disable it, i can join servers again, just wanted to let you know

**Comment by sofiadif** [comment]
@PepinCZ: Also tried your raw file, 2 x 45minutes.. Worked awesome for me :)

**Comment by kaue** [comment]
@PepinCZ Awesome! so the problem was with the `mapLeft` and `mapTop`? 

I think using `Math.abs` is still a temp fix, anyone noticed any strange stuff with the minimap after this change?

**Comment by ghost** [comment]
@kauegimenes no, its it does not matter if its `mapLeft` and `MapTop` or `MapRight` and `mapBottom`... the problem was `blob.size / minimapScale` goin into negative value, so `Math.abs` solved the issue. And I dont think its temp fix, because that the minimap uses to not fall into negative value.

**Comment by ghost** [comment]
@Aers20 I think that was pure accident.. its not possible. I tested it.

**Comment by mrblah1** [comment]
Anybody else getting the awwww snap crash error in chrome now?

**Comment by TheCranky** [comment]
Yeah I sometimes get it it might be because of not enough ram tho

**Comment by ghost** [comment]
@mrblah1 hmm, thats weird.. please tell me your system information (cpu, ram...)

**Comment by mrblah1** [comment]
Corei5 3.5Ghz and 16GB of ram with win7.

**Comment by mrblah1** [comment]
I'm also experiencing another glitch where a split blob just gets stuck and can't move anymore, my other blobs can move but that one stays put forever, nobody else can eat it either.  Anybody else experiencing this?

**Comment by ghost** [comment]
@mrblah1 ok, thats weird... i not getting any of thsi bugs...

**Comment by kaue** [comment]
@mrblah1 Can you confirm this only happens with ZeachCobbler bot?

**Comment by sofiadif** [comment]
mrblah1: I'm having the same problem, a few times now, randomly.
I never tried without ZeachCobbler but i've never seen the problem on /r/agario.
So my guess, its ZC

**Comment by sofiadif** [comment]
I'm also having the issue as @mrblah1 
I haven't tried playing without ZC, but also never seen anyone having the bug on /r/agario.

So my guess - maybe its ZC but i can't be sure. It happens randomly a few times..

I've also seen other smaller blobs being "afk", but won't be eaten. And I'm not sure other players can see them

My blob never disappears from the screen, if its stuck in a corner, and i'll go to the other corner, i would see the whole map
Here's a video: https://www.youtube.com/watch?v=EOXjCWM9KlA

**Comment by ghost** [comment]
@sofiadif @mrblah1 When I testing many times I have issue that some player(s) on server get stucked and I cant eat them and they are stuck in the minimap forever.... This also happened to me with food pallets. But never when I split up. Hey Its a weird glitch. Anyone please help, because i dont know what is this.

**Comment by jon-bullpoint** [comment]
The problem is DebugMonkey didn't document the protocol, so when the
protocol changes, it's impossible to easily decipher which bytes are
new/changed even for an experienced programmer.

I'm not willing to reverse the agar.io protocol, so it's unfortunate that I
can't help.

**Comment by ghost** [comment]
I have the same issue so I open another bug copy/pasting @sofiadif comment with the self explanatory video.

**Comment by Basharb** [comment]
When i open agar.io it freezes it's like stuck

Check this >> http://prnt.sc/b6ggvg
